### PR TITLE
TER-001 Create Bash runner

### PR DESCRIPTION
**Pull request by nashikhmin**

### CODE REVIEW

File: src/main/kotlin/ru/jetbrains/testenvrunner/utils/BashExecutor.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+         * @return the result of bash script executing
+         */
+        fun executeCommand(command: ExecutionCommand, envParams: Map<String, String> = emptyMap(), directory: String = "", waitingTime: Long = DEFAULT_WAITING_TIME): ExecutionResult {
+            val pb: ProcessBuilder = ProcessBuilder(command.command.split(" "))
+            pb.redirectErrorStream(true)
+            if (!directory.isEmpty()) {
+                pb.directory(File(directory))
+            }
+
+            var p: Process? = null
```
Comment (nashikhmin) [inline src/main/kotlin/ru/jetbrains/testenvrunner/utils/BashExecutor.kt:34]: Можно ли в котлин как-то без этого?

File: .gitignore [hunk 2]
```diff
@@ -21,4 +21,4 @@ build/
 nbbuild/
 dist/
 nbdist/
-.nb-gradle/
\ No newline at end of file
+.nb-gradle/
```
Comment (mkuzmin) [inline .gitignore:24]: 👍

File: documentation/decisions/TER-000 Ideas.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+In progress
+
+## Context
+
+We need to create the application to 31/08/2017. 
+
+## Decision
+
+- We concentrate on features of the project, not on the visual design.
```
Comment (mkuzmin) [inline documentation/decisions/TER-000 Ideas.md:15]: need to discuss this tomorrow.
ADRs are for used technologies only. Features and project workflows need to be documented somewhere else

File: documentation/decisions/TER-000 Ideas.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+- We use Kotlin as the language.
+- We use Spring Boot as the framework.
```
Comment (mkuzmin) [inline documentation/decisions/TER-000 Ideas.md:17]: let's split each decision into separate file

File: documentation/decisions/TER-000 Ideas.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+- We use Mongodb as the database.
```
Comment (mkuzmin) [inline documentation/decisions/TER-000 Ideas.md:18]: may postpone this decision until we actually start working on related feature request

File: src/main/kotlin/ru/jetbrains/testenvrunner/TestEnvrunnerApplication.kt [hunk 6]
```diff
@@ -1,11 +1,11 @@
-package ru.jetbrains.testenvrunner
+package ru.jetbrains
```
Comment (mkuzmin) [inline src/main/kotlin/ru/jetbrains/testenvrunner/TestEnvrunnerApplication.kt:1]: at the moment this does not matter much, but we do need an unique name here. Better under `org.jetbrains`

File: src/main/resources/templates/index.html [hunk 7]
```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org" lang="en"
+      xmlns="http://www.w3.org/1999/xhtml">
+<head>
+    <meta charset="UTF-8"/>
+    <title>Title</title>
+
+    <!-- Latest compiled and minified CSS -->
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css"
```
Comment (mkuzmin) [inline src/main/resources/templates/index.html:9]: We'll probably need a separate ADR for bootstrap

File: documentation/decisions/TER-000 Ideas.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
+# 1. Create Test Environment Runner
```
Comment (mkuzmin) [inline documentation/decisions/TER-000 Ideas.md:1]: ADRs can be posted right into `develop` - depends from specific context

File: src/main/resources/templates/result.html [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+            integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa"
+            crossorigin="anonymous"></script>
+</head>
+<body>
+<div class="container">
+    <div class="row">
+        <div class="col-xs-12 col-sm-12 col-md-12 col-lg-12">
+            <h1> The result of command execution</h1>
+            <!--/*@thymesVar id="result" type="ru.jetbrains.testenvrunner.model.ExecutionResult"*/-->
+            <p th:text="${result.result}"></p>
```
Comment (mkuzmin) [inline src/main/resources/templates/result.html:29]: need to handle line endings
image

File: pom.xml [hunk 10]
```diff
@@ -1,115 +1,115 @@
...
-		</plugins>
-	</build>
+        <dependency>
+            <groupId>io.github.microutils</groupId>
+            <artifactId>kotlin-logging</artifactId>
+            <version>1.4.5</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.jetbrains.spek</groupId>
```
Comment (mkuzmin) [inline pom.xml:62]: We'll probably will use Spek for Selenium tests, but not now

Comment (nashikhmin) [reply pom.xml:62]: ok

File: src/main/kotlin/ru/jetbrains/testenvrunner/TestEnvrunnerApplication.kt (hunk 6 above)
Comment (nashikhmin) [reply src/main/kotlin/ru/jetbrains/testenvrunner/TestEnvrunnerApplication.kt:1]: ok, rename,when will be move to official repo

File: src/main/resources/templates/result.html (hunk 9 above)
Comment (nashikhmin) [reply src/main/resources/templates/result.html:29]: ok, improved

**Pull request closed without merging**